src/storage/src/hummock/sstable_store.rs

```diff
@@ -372,6 +224,63 @@ impl SstableStore {
     pub fn clear_block_cache(&self) {
         self.block_cache.clear();
     }
+
+    pub async fn load_table(
+        &self,
+        sst_id: HummockSSTableId,
+        stats: &mut StoreLocalStatistic,
+        load_data: bool,
+    ) -> HummockResult<TableHolder> {
+        loop {
+            stats.cache_meta_block_total += 1;
+            let entry = self
+                .meta_cache
+                .lookup_with_request_dedup::<_, HummockError, _>(sst_id, sst_id, || async {
+                    stats.cache_meta_block_miss += 1;
+                    let path = self.get_sst_meta_path(sst_id);
+                    let buf = self
+                        .store
+                        .read(&path, None)
+                        .await
+                        .map_err(HummockError::object_io_error)?;
+                    let mut size = buf.len();
+                    let meta = SstableMeta::decode(&mut &buf[..])?;
+                    let sst = if load_data {
+                        size = meta.estimated_size as usize;
+                        let data_path = self.get_sst_data_path(sst_id);
+                        let block_data = self
+                            .store
+                            .read(&data_path, None)
+                            .await
+                            .map_err(HummockError::object_io_error)?;
+                        Sstable::new_with_data(sst_id, meta, block_data)?
+                    } else {
+                        Sstable::new(sst_id, meta)
+                    };
+                    Ok((Box::new(sst), size))
+                })
+                .await
+                .map_err(|e| {
+                    HummockError::other(format!(
+                        "meta cache lookup request dedup get cancel: {:?}",
+                        e,
+                    ))
+                })??;
+            if !load_data || !entry.value().blocks.is_empty() {
+                return Ok(entry);
+            }
+            drop(entry);
```

Comment:
In what case will we reach this code? When the physical sst is empty and has no block?

Reply:
If some other thread cache this sst in meta-cache but it does not cache block data in `Sstable`, we shall read this meta again.